src/main/test/surfrdUtils_test/test_surfrdUtils.pf

```diff
@@ -13,15 +13,402 @@ module test_surfrdUtils
 
 contains
   
+  @Test
+  subroutine test_collapse_crop_types_none()
```

Comment:
I'm glad you have all these different tests here. They seem to be comprehensive and extensive. There are two things I wonder about. The first thing I wonder about is that it looks like there is quite a bit of setup that's similar if not identical between them all. So you might be able to simplify it by having a shared setup subroutine. The other thing to think about is that the checks at the end are all hardcoded to match the setup at the beginning, meaning that it's not totally obvious that it matches. A goal of unit-testing is to try to make it obvious that it's doing the right thing. If it could be made more obvious that would be helpful. One way to do that would be to have the "expect" array settings based on the input settings. So then it's clear what part is supposed to be the same, and what part is different because it's should be.

Reply:
This comment will take a little thought to address, so I will handle it after I commit the simpler revisions.

Reply:
Now added "expect" array settings as recommended.

Reply:
And next added subroutine setup_test_collapse_crop_types as recommended. (Unit tests continue to PASS.)

Reply:
Excellent. This looks much better and so much easier to read! The setup subroutine moves off the busy work, so you just see the core of what's happening.

I have one more suggestion if you are willing to hang on for just a little more...

You set the input arrays element by element and then do the same for the expected arrays. I think it would be easier to see if you did this using array syntax, and also set the expected arrays based on the input arrays (since typically there is some simple change that is going on). For example, in the 20_to_18 test you have...

``` fortran
     ! Input values
     wt_cft(begg:,1) = 5.0_r8
     wt_cft(begg:,2) = 10.0_r8
     wt_cft(begg:,3) = 20.0_r8
     wt_cft(begg:,4) = 30.0_r8
     wt_cft(begg:,5) = 15.0_r8
     wt_cft(begg:,6) = 20.0_r8
     call check_sums_equal_1( (wt_cft/100.0_r8), begg, "test_check_sums_add_to_1", &
                              "should not trigger an error")
     fert_cft(begg:,1) = 5.0_r8
     fert_cft(begg:,2) = 10.0_r8
     fert_cft(begg:,3) = 20.0_r8
     fert_cft(begg:,4) = 30.0_r8
     fert_cft(begg:,5) = 15.0_r8
     fert_cft(begg:,6) = 20.0_r8
      ! Output values expected
     wt_cft_expected(begg:,1) = 5.0_r8
     wt_cft_expected(begg:,2) = 10.0_r8
     wt_cft_expected(begg:,3) = 0.0_r8
     wt_cft_expected(begg:,4) = 0.0_r8
     wt_cft_expected(begg:,5) = 35.0_r8
     wt_cft_expected(begg:,6) = 50.0_r8
```

Here's what I suggest you change it to...

``` fortran
     ! Input values
     wt_cft(begg:,:) = (/ 5.0_r8, 10.0_r8, 20.0_r8, 30.0_r8, 15.0_r8, 20.0_r8 /)
     call check_sums_equal_1( (wt_cft/100.0_r8), begg, "test_check_sums_add_to_1", &
                              "should not trigger an error")
     fert_cft(begg:,:) = wt_cft(begg:,:)

      ! Output values expected, same as before with 3 and 4 zeroed, but added into 5 and 6
     wt_cft_expected(begg:,:) = wt_cft(begg:,:)
     wt_cft_expected(begg:,3:4) = 0.0_r8
     wt_cft_expected(begg:,5) = wt_cft(begg:3) + wt_cft(begg:5)
     wt_cft_expected(begg:,6) = wt_cft(begg:4) + wt_cft(begg:6)
```

When I read the unit test, to see what's going on I have to scan through all the values, just to realize they are duplicated from what was before. Since, most of them are duplicated it helps to just see which ones change. Technically, that would also allow you to change the values in the original setting of wt_cft and the unit test would still work. But, the main thing I'm thinking about is readability.

A goal of a unit test is to make it as readable as possible, so you don't have to think about it much to parse what it's doing. If you can just scan it and see that it's correct that's ideal, but often hard to achieve. But, working toward that is desirable. Looking at the above the second one is much shorter and I would say easier to read and see what it's doing. Hopefully, that makes sense.

Reply:
Will do. I will commit and push by the end of today.

Reply:
Done.